### DISCUSSION TITLE
Welcome to ZUGFeRD-for-Delphi Discussions!

### DISCUSSION THREAD

**Discussion by landrix** [LandrixSoftware/ZUGFeRD-for-Delphi#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Ricoarlo** [comment]
Hello,

Thanks for your work ;)
I try to make a french invoice FacturX with your library and I have questions.
In unit intf.ZUGFeRDSubjectCodes, why you don't have added all codes from http://www.stylusstudio.com/edifact/D02A/4451.htm ?
In my case I need codes PMD and AAB

In France we have a tax named Ecotaxe (DEEE) based on quantity, I don't find how to record it.
This tax has a fixed price and has TVA
Must I record it as a product ?

Have plan something to localise these functions ? :
function TZUGFeRDInvoiceDescriptor23CIIWriter._translateTaxCategoryCode(taxCategoryCode : TZUGFeRDTaxCategoryCodes) : String;
function TZUGFeRDInvoiceDescriptor22CIIWriter.translateInvoiceType(type : TZUGFeRDInvoiceType) : String;

In french invoice I can't have german ;)

When I add a comment with AddTradeLineCommentItem, when I check the invoice with https://services.fnfe-mpe.org/account/home : I have this error : XML file invalid against XSD
Not sure it's possible to add a comment as a product without specification ...

Thank you
Eric

**Comment by landrix** [comment]
> In France we have a tax named Ecotaxe (DEEE) based on quantity, I don't find how to record it.
This tax has a fixed price and has TVA
Must I record it as a product ?

do you have a example in xml? or can you show me infos about that in factur-x documentation?
